### PR TITLE
[Feature] Support Alter Catalog Statement

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 15 / 86 (17.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[303]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AuthorizerStmtVisitor.java|0|2|00.00%|[546, 548]|
|:large_blue_circle:|com/starrocks/persist/AlterCatalogLog.java|0|9|00.00%|[34, 35, 36, 37, 40, 44, 49, 50, 53]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterCatalogStmt.java|0|7|00.00%|[23, 24, 25, 26, 29, 33, 38]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|6|00.00%|[985, 990, 991, 992, 1962, 1963]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CatalogAnalyzer.java|0|9|00.00%|[123, 124, 125, 126, 128, 129, 130, 132, 134]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|3|00.00%|[960, 961, 962]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|0|4|00.00%|[854, 855, 856, 857]|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|5|33|15.15%|[166, 167, 169, 170, 171, 174, 175, 176, 177, 178, 180, 183, 185, 186, 189, 191, 261, 294, 296, 297, 298, 299, 300, 302, 305, 306, 308, 310]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerVisitor.java|10|12|83.33%|[545, 546]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 73 / 86 (84.88%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[303]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|3|00.00%|[960, 961, 962]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|2|6|33.33%|[985, 990, 991, 992]|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|29|33|87.88%|[171, 178, 261, 300]|
|:large_blue_circle:|com/starrocks/persist/AlterCatalogLog.java|8|9|88.89%|[53]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerVisitor.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AuthorizerStmtVisitor.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterCatalogStmt.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CatalogAnalyzer.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by gengjun-git** [comment]
Should add alter operation into SQLTest

File: fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java [hunk 1]
```diff
@@ -250,6 +290,25 @@ public void replayDropCatalog(DropCatalogLog log) {
         }
     }
 
+    public void replayAlterCatalog(AlterCatalogLog log) {
+        writeLock();
+        try {
+            String catalogName = log.getCatalogName();
+            Map<String, String> properties = log.getProperties();
+            String serviceName = properties.get("ranger.plugin.hive.service.name");
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java:298]: Use a constant replace "ranger.plugin.hive.service.name"?

**Comment by stephen-shelby** [comment]
Alter catalog equals Drop + create catalog. you need to clear all client information and cache info when you modify the catalog properties. for example, if user update the hive.metastore.uris, the hive meta client need to recreate using new urls. So it's better to drop first and then create catalog. you could refers to the alter resource.

File: fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java [hunk 2]
```diff
@@ -159,6 +162,34 @@ public void dropCatalog(DropCatalogStmt stmt) {
...
+                if (serviceName.isEmpty()) {
+                    Authorizer.getInstance().setAccessControl(catalogName, new NativeAccessControl());
+                } else {
+                    Authorizer.getInstance().setAccessControl(catalogName, new RangerHiveAccessControl(serviceName));
+                }
+
+                catalog.getConfig().put("ranger.plugin.hive.service.name", serviceName);
+
+                AlterCatalogLog alterCatalogLog = new AlterCatalogLog(catalogName, properties);
+                GlobalStateMgr.getCurrentState().getEditLog().logAlterCatalog(alterCatalogLog);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java:186]: clear connector info.

File: fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java [hunk 3]
```diff
@@ -159,6 +162,34 @@ public void dropCatalog(DropCatalogStmt stmt) {
         }
     }
 
+    public void alterCatalog(AlterCatalogStmt stmt) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java:165]: nit: Add a TODO to support modify other properties

**Review by stephen-shelby — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by wangsimo0 — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
@Mergifyio backport branch-3.1

**Comment by HangyuanLiu** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**